feature/rds/auth/doc.go

```diff
@@ -0,0 +1,7 @@
+// Package auth is used to generate authentication tokens used to
+// connect to a givent Amazon Relational Database Service (RDS) database.
+//
+// Before using the authentication please visit the docs here to ensure
+// the database has the proper policies to allow for IAM token authentication.
+// https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.html#UsingWithRDS.IAMDBAuth.Availability
```

Comment:
Nit: we can point to our developer docs too - since they are more helpful wrt usage in SDK .